test/ocsp_peer_test.go

```diff
@@ -0,0 +1,2875 @@
+				}
+			}`,
+			nil,
+			nil,
+		},
+		{
+			"No response in cache for UserA1 -- should be allowed connection",
+			`
+				port: -1
+				http_port: 8222
+				tls: {
+					cert_file: "configs/certs/ocsp_peer/mini-ca/server1/TestServer1_bundle.pem"
+					key_file: "configs/certs/ocsp_peer/mini-ca/server1/private/TestServer1_keypair.pem"
+					ca_file: "configs/certs/ocsp_peer/mini-ca/root/root_cert.pem"
+					timeout: 5
+					verify: true
+					# Turn on CA OCSP check but allow when CA is unreachable
+					ocsp_peer: {
+						verify: true
+						ca_timeout: 0.5
+						allow_when_ca_unreachable: true
+					}
+				}
+				# preserve revoked true
+				ocsp_cache: {
+					type: local
+					preserve_revoked: true
+				}
+			`,
+			[]nats.Option{
+				nats.ClientCert("./configs/certs/ocsp_peer/mini-ca/client1/UserA1_bundle.pem", "./configs/certs/ocsp_peer/mini-ca/client1/private/UserA1_keypair.pem"),
+				nats.RootCAs("./configs/certs/ocsp_peer/mini-ca/root/root_cert.pem"),
+				nats.ErrorHandler(noOpErrHandler),
+			},
+			"",
+			nil,
+			nil,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			deleteLocalStore(t, "")
+			c := []byte(test.cachedResponse)
+			err := writeCacheFile("", c)
+			if err != nil {
+				t.Fatal(err)
+			}
+			content := test.config
+			conf := createConfFile(t, []byte(content))
+			s, opts := RunServerWithConfig(conf)
+			defer s.Shutdown()
+			nc, err := nats.Connect(fmt.Sprintf("tls://localhost:%d", opts.Port), test.opts...)
+			if test.err == nil && err != nil {
+				t.Errorf("Expected to connect, got %v", err)
+			} else if test.err != nil && err == nil {
+				t.Errorf("Expected error on connect")
+			} else if test.err != nil && err != nil {
+				// Error on connect was expected
+				if test.err.Error() != err.Error() {
+					t.Errorf("Expected error %s, got: %s", test.err, err)
+				}
+				return
+			}
+			defer nc.Close()
+		})
+	}
+}
+
+// TestOCSPResponseCacheLocalStoreOption is test of default and non-default local_store option
+func TestOCSPResponseCacheLocalStoreOpt(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	rootCAResponder := newOCSPResponderRootCA(t)
+	rootCAResponderURL := fmt.Sprintf("http://%s", rootCAResponder.Addr)
+	defer rootCAResponder.Shutdown(ctx)
+	setOCSPStatus(t, rootCAResponderURL, "configs/certs/ocsp_peer/mini-ca/intermediate1/intermediate1_cert.pem", ocsp.Good)
+
+	for _, test := range []struct {
+		name           string
+		config         string
+		opts           []nats.Option
+		cachedResponse string
+		err            error
+		rerr           error
+		storeLocation  string
```

Comment:
It's probably a good idea to make use of [t.TempDir](https://pkg.go.dev/testing#B.TempDir) so that the testing framework sets up temporary directories and cleans them up automatically, rather than potentially polluting the current working directory.

Reply:
Yes, get your point.  A bit tricky here because the test is verifying the default location of OCSP Response Cache when `local_store` is unset which is relative `_rc_` (not a temp location).   I modeled this after JetStream default store of `_js_`.